ingest-app/src/cmr/ingest/api/core.clj

```diff
@@ -61,7 +61,7 @@
     :result
     {}
     (reduce-kv (fn [memo k v]
-                 (conj memo (xml/element (keyword k) {} v)))
+                 (conj memo (xml/element (keyword k) {} (if (map? v) (str v) v))))
```

Comment:
I want to ask why this is a needed change? Will this result in json like text in the XML? This is a private function, so was this needed to fix a bug?

Maybe a better example would help?

Reply:
This change is needed because now variable ingest returns not only concept-id and revision-id, it also returns variable association and associated-item, both of them are map. There is no requirement on how they need to be returned, so I'm just returning them as strings.
<?xml version="1.0" encoding="UTF-8"?><result><concept-id>V1200000007-PROV1</concept-id><revision-id>1</revision-id><variable-association>{:concept-id "VA1200000008-CMR", :revision-id 1}</variable-association><associated-item>{:concept-id "C1200000005-PROV1"}</associated-item></result>

Reply:
Looks like what is returned is a mix of XML and JSON. If I request JSON to be returned, will I get all properly formatted JSON returned?

Reply:
{"concept-id":"V1200000012-PROV1","revision-id":1,"variable-association":{"concept-id":"VA1200000007-CMR","revision-id":1},"associated-item":{"concept-id":"C1200000005-PROV1","revision-id":1}}

Reply:
I think the mixed content is a problem, don't we have a method to convert it to some generic xml, like <key>value</key>?

Reply:
Sure. Now that I got the main functionality working, I can go back to take care of that. Thanks.

Reply:
Or you can flatten it out as son: 
{"concept-id":"V1200000012-PROV1","revision-id":1,"variable-association-concept-id":"VA1200000007-CMR","variable-association-revision-id":1,"associated-item-concept-id":"C1200000005-PROV1","associated-itemrevision-id":1}
as XML:
<?xml version="1.0" encoding="UTF-8"?><result><concept-id>V1200000007-PROV1</concept-id><revision-id>1</revision-id><variable-association-concept-id>VA1200000008-CMR</<variable-association-concept-id><variable-association-revision-id>1</variable-association-revision-id><associated-item-concept-id>C1200000005-PROV1</associated-item-concept-id></result>

Reply:
<?xml version="1.0" encoding="UTF-8"?><result>
    <concept-id>V1200000006-PROV1</concept-id>
    <revision-id>2</revision-id>
    <variable-association>
        <concept-id>VA1200000007-CMR</concept-id>
        <revision-id>2</revision-id>
    </variable-association>
    <associated-item>
        <concept-id>C1200000005-PROV1</concept-id>
        <revision-id>1</revision-id>
    </associated-item>
</result>